terraform/modules/firestore/main.tf

```diff
@@ -27,3 +27,21 @@ resource "google_project_iam_member" "scaler_sa_firestore" {
   role    = "roles/datastore.user"
   member  = "serviceAccount:${var.scaler_sa_email}"
 }
+
+resource "google_project_service" "firestore" {
+  count = var.terraform_firestore_create ? 1 : 0
+
+  project = var.project_id
+  service = "firestore.googleapis.com"
+}
+
+resource "google_firestore_database" "database" {
+  count = var.terraform_firestore_create ? 1 : 0
```

Comment:
Since this is in the Terraform module for Firestore, wouldn't using the module indicate we want to create the Firestore database.  Also choosing to not create the database while using the module would break the setup, correct?

Reply:
Hello,

thanks for the review,
I sort of agree with you but if we do not provide an option then we will end up forcing the user to use Firestore. What in case if the user wants to use Spanner to store the Autoscaler state?
Should we still create the Firestore db as a default setup and let the user use Spanner without utilizing the provisioned Firestore db?

Let me know if my understanding is not accurate.

Thanks!

Reply:
Previously, if the user wanted to store state in Spanner the documentation instructed them to skip the gcloud command to create the Firestore database.  Now with the module users in that situation should just not include the Firestore module in their Terraform.